rf2222222/Provenance
# Play: Cheat Code Support

Merged: https://github.com/Provenance-Emu/Provenance/pull/2089

### What does this PR do
   As a natural course of playing playstation 2, the need for entering cheat code happens. Play Core doesn't have a native support for cheat codes, so this will write the cheat directly to EE->m_ram. In addition to raw address -> value pairs, Code Breaker, Pro Action Replay v1/v2, and Game Genie cheat codes are supported (porting omniconvert's game code->raw conversion utility in PVPlayCore+Cheats.mm 

This PR implements the following:
  * Updates the Cheats Storyboard to support selection for Cheat Code Type (e.g. Game Genie, etc) in both TV / iOS
  * Updates GameWithCheat protocol to setCheat with CheatCodeType
  * Implements Cheat Support (PVPlayCore+Cheats extension) on Play! Core using above, with support for Code Breaker/Pro Action Replay V1 V2/Game Genie and Raw codes (adds lib big int needed for cheat code -> raw conversion / decryption)

### Where should the reviewer start
   * PVPlayCore/Core/PVPlayCore+Cheats 
   * Cheats Storyboards
   * GameWithCheat protocol

### How should this be manually tested
   * Load PS2 Game -> Apply Cheat Code -> Cheat code should work (invalid cheat code will cause crash)

### Any background context you want to provide
   * Needed cheat codes while playing PS2 game, figured it's faster to implement the feature

### What are the relevant tickets

### Screenshots (important for UI changes)

### Questions
